File: Land/Grassland.cpp
/**
 * @file Grassland.hpp
 * @author 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @brief file berisi definisi class Grassland
 * @version 0.1
 * @date 2019-03-20
 */

/* NIM/Nama  : 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * Nama file : Grassland.hpp
 * Topik     : Tugas Besar 1 IF2210 - Pemograman Berorientasi Objek
 * Tanggal   : 20 Maret 2019
 * Deskripsi : Header Grassland.cpp */

#include "Grassland.hpp"
#include <random>
#include <iostream>

using namespace std;

/**
 * @brief Construct a new Land object
 * 
 */
Grassland::Grassland(){
    random_device dev;
    mt19937 rng(dev());
    uniform_int_distribution<mt19937::result_type> dist6(0,8);

    grassStatus = dist6(rng);
    if(grassStatus != 0)
    	grassStatus = 0;
    else
    	grassStatus = 1;
}

/**
 * @brief Return true if there is grass
 * 
 * @return int 
 */
int Grassland::isGrass(){
	return grassStatus;
}

/**
 * @brief Set the Grass Status object
 * 
 */
void Grassland::setGrassStatus(int gs){
	grassStatus = gs;
}

/**
 * @brief Method to render the land type character to map
 * 
 * @return char 
 */
char Grassland::render(){
	if(!isGrass())
		return '-';
	else
		return '#';
}

File: Java/model/land/Grassland.java
/**
 * Grassland.java
 * <p>
 * File berisi class Grassland
 * </p>
 * @author 13515125/Muhammad Rafid Amrullah, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @version 2.0
 * @since 2019-04-21
 */

package land;

import resource.*;
import java.io.*;
import java.lang.*;
import java.util.*;

/**
 * <p>
 * Class for rendering object to Grassland
 * </p>
 */
public class Grassland implements resource.Cell
{
	private boolean grassStatus;

	/**
	 * Grassland constructor
	 */
	public Grassland(){
		Random random = new Random();
		int n = random.nextInt(5);
		grassStatus = (n == 0);
	}
	
	/**
	 * Get the object grassStatus
	 * @return boolean
	 */
	public boolean isGrass(){
		return grassStatus;
	}

	/**
	 * Set the object grassStatus
	 * @param grassStatus
	 */
	public void setGrassStatus(boolean grassStatus){
		this.grassStatus = grassStatus;
	}

	/**
	 * Render grassland to map
	 * @return Character
	 */
	public Character render(){
		if(!isGrass())
			return '-';
		else
			return'#';
	}
}
